### ISSUE TITLE
Improve Java Runtime Environment Selection

### ISSUE THREAD

**Issue by LostDragonist** [ModOrganizer2/modorganizer#672]
Reference: 
https://github.com/ModOrganizer2/modorganizer/blob/583bc5b6c669bb0496b1818497ce62deb562ddc0/src/editexecutablesdialog.cpp#L167

This code should probably be changed to heavily encourage the user to use a 64-bit JRE.  This can avoid issues with certain utilities like the SkyProc patcher (dual sheath redux, etc.). 

For example, when a jar is selected, a dialog could be popped up asking whether to use the detected 64-bit JRE or the detected 32-bit JRE.  A message would be present that suggests to use a 64-bit JRE unless absolutely required.  If either JRE is not detected, provide a link to where the user can download that JRE.  

At the very least, the message about "MO requires 32-bit java" needs to be changed as it's not really true.  

Make sure the behavior of the data list "add as executable" is consisent with the edit executables dialog.

**Comment by LostDragonist** [comment]
It might be overstepping a bit but also consider automatically adding `-Xmx1024x` arguments to any usage of the 32-bit JRE.  Maybe even suggest adding `Xmx2048x` arguments to any usage of the 64-bit JRE.

These appear to have positive effects for SkyProc patchers but may have negative effects for other utilities...?

**Comment by SuperSandro2000** [comment]
Why should it have negative effects? Only if the user has little to no memory free.

**Comment by LostDragonist** [comment]
I was thinking if the executable somehow has a higher memory limit defined and the arguments override it to a lower memory limit?... I have no clue of that makes sense (as with most things surrounding Java).

**Comment by SuperSandro2000** [comment]
The flag is an arg to the jvm (java virtual machine) and as far as my basic knowledge can remember the application can't change that.
Just add the flag and tweak it if necessary.

**Comment by AnyOldName3** [comment]
I just googled the default maximum heap size for the 64-bit JVM and came up with a quarter of physical memory. Are we sure we want to go passing the JVM options to change it?

By the way, I'm not seeing what the `x` suffix means. I can see in the docs that no suffix is bytes, `k` is kilobytes and `m` is megabytes, but not a multiplier for `x`. I doubt it's gigabytes, as there's no way a 32-bit JVM is letting you allocate a terabyte of heap.

**Comment by LostDragonist** [comment]
It looks like I was contributing to some internet hearsay about that.  I just copied that off some forum about the issue.  Oracle Java refuses to run with that command at all for me.  

The proper commands, per the batch files included with Dual Sheath Redux, should be `-Xmx1024m` and `-Xmx2048m`.  

My default heap size is 256MB for my 32-bit JVM and ~4GB for my 64-bit JVM.  (32GB of RAM; determined with `.\java -XX:+PrintFlagsFinal -version | findstr HeapSize`)

So... it's probably worth leaving the 64-bit JVM alone unless something else happens.

**Comment by SuperSandro2000** [comment]
I agree with you. Setting it to high will cause the GC to lag. We can already change the vars. Don't think there are any changes required.
